### PR TITLE
Filter by date range backend

### PR DESCRIPTION
**Pull request by TeerapatTrepopsakulsin**
### Description
- Implement filter by day of week backend #163

### Functional Acceptance Criteria
- `GET activities/?start_date=<date>&end_date=<date>` Return list of activities that take time that in between start_time and end_time (_Both **Inclusive**_). More details about the format in additional information.

### Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation, if applicable.
- [x] My code follows the Coding Guidelines.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other (please specify)

### Additional Information
date format: `r'(?P<year>\d{4})-(?P<month>\d{1,2})-(?P<day>\d{1,2})$'`, for example, 2024-01-20.
Example: 
`GET activities/?start_date=2024-10-10` Will return a list of activities from 10 Oct 2024 on.
`GET activities/?start_date=2024-10-15&end_date=2024-10-20` Will return a list of activities from 15 Oct 2024 until 20 Oct 2024 (Inclusive).

### CODE REVIEW

**Comment by Thanawas-Sirilertsathit** [comment]
I already give you a comment. @TeerapatTrepopsakulsin

**Comment by Thanawas-Sirilertsathit** [comment]
Please clarify

**Review by OmegaOoh — APPROVED** [review]
Looks fine

File: activities/tests/test_index.py
Comment (Thanawas-Sirilertsathit) [reply activities/tests/test_index.py:229]: I got answer already ok

**Review by Thanawas-Sirilertsathit — APPROVED** [review]
Your answer for me is reasonable. I approved.

**Review by Thanawas-Sirilertsathit — APPROVED** [review]
Approve everything is fine after you merged

**Pull request merged**
